src/components/Tokens/TokenDetails/ChartSection.tsx

```diff
@@ -51,55 +52,52 @@ const TokenActions = styled.div`
   gap: 16px;
   color: ${({ theme }) => theme.textSecondary};
 `
-const NetworkBadge = styled.div<{ networkColor?: string; backgroundColor?: string }>`
-  border-radius: 5px;
-  padding: 4px 8px;
-  font-weight: 600;
-  font-size: 12px;
-  line-height: 12px;
-  color: ${({ theme, networkColor }) => networkColor ?? theme.textPrimary};
-  background-color: ${({ theme, backgroundColor }) => backgroundColor ?? theme.backgroundSurface};
-`
 
-export default function ChartSection({ token, tokenData }: { token: Token; tokenData: SingleTokenData | undefined }) {
-  const { chainId: connectedChainId } = useWeb3React()
+export function useTokenLogoURI(
+  token: NonNullable<SingleTokenData> | NonNullable<TopToken>,
+  nativeCurrency?: Token | NativeCurrency
+) {
+  const checksummedAddress = isAddress(token.address)
+  const chainId = CHAIN_NAME_TO_CHAIN_ID[token.chain]
+  return (
+    useCurrencyLogoURIs(nativeCurrency)[0] ??
+    (checksummedAddress && getTokenLogoURI(checksummedAddress, chainId)) ??
+    token.project?.logoUrl
+  )
+}
+
+export default function ChartSection({
+  token,
+  nativeCurrency,
+}: {
+  token: NonNullable<SingleTokenData>
+  nativeCurrency?: Token | NativeCurrency
+}) {
   const isFavorited = useIsFavorited(token.address)
   const toggleFavorite = useToggleFavorite(token.address)
-  const chainInfo = getChainInfo(token?.chainId)
-  const networkLabel = chainInfo?.label
-  const networkBadgebackgroundColor = chainInfo?.backgroundColor
-  const warning = checkWarning(token.address)
-
-  let currency = useCurrency(token.address)
+  const chainId = CHAIN_NAME_TO_CHAIN_ID[token.chain]
+  const L2Icon = getChainInfo(chainId).circleLogoUrl
+  const warning = checkWarning(token.address ?? '')
```

Comment:
why did you add the `?? ''` here? it looks like the func signature didn't change

Reply:
token was a Token() before as in a result of useToken(), now its just token data from the api. every element from the api is potentially null, but I'm planning at looking into the @required directive to see if we can get rid of some repetitive ternarys and nullchecking https://relay.dev/docs/next/guides/required-directive/